### PR TITLE
KPMP-4614_add-release

### PR DESCRIPTION
**Pull request by Dert1129**

### CODE REVIEW

File: src/main/resources/knowledge_environment.graphqls [hunk 1]
```diff
@@ -30,6 +30,7 @@ type SpatialViewerDataset {
     level: String # may be null
     relatedFiles: [File] # may be null
     externalLink: String # may be null
+    releaseVersion: Float
```
Comment (rlreamy) [inline src/main/resources/knowledge_environment.graphqls:33]: Shouldn't this be a String?

File: src/main/java/org/kpmp/spatialViewerDataset/SpatialViewerDatasetService.java [hunk 2]
```diff
@@ -59,10 +65,25 @@ public SpatialViewerDatasetService(
...
+    List <SpatialViewerDataset> datasetsFinal = new ArrayList<>();
+    List <SpatialViewerFileDataset> datasets = new ArrayList<>();
+    Map<String, SpatialViewerFileDataset> fileMap = new HashMap<>();
+    Double maxReleaseVersion = fileRepo.max();
     datasets.addAll(fileRepo.findAll());
-    return datasets;
+    for (SpatialViewerFileDataset spatialViewerFileDataset : datasets){
+        if(spatialViewerFileDataset.getReleaseVersion() != null){
+            if (Double.compare(spatialViewerFileDataset.getReleaseVersion(), maxReleaseVersion) == 0){
+                spatialViewerFileDataset.setReleaseVersionDisplay("Recently Released");
```
Comment (rlreamy) [inline src/main/java/org/kpmp/spatialViewerDataset/SpatialViewerDatasetService.java:76]: This should be extracted to a constant. Remember constants should be `private static final`, and live at the top of the file with other class variables.

File: src/main/java/org/kpmp/spatialViewerDataset/SpatialViewerDatasetService.java [hunk 3]
```diff
@@ -48,6 +50,10 @@ public void setErrors(List errors) {
 		}
 	}
 
+    public SpatialViewerDatasetService() {
```
Comment (rlreamy) [inline src/main/java/org/kpmp/spatialViewerDataset/SpatialViewerDatasetService.java:53]: Spring should have nothing to complain about. You should be able to have an overridden constructor with parameters without defining a default constructor.

**Review by rlreamy — CHANGES_REQUESTED** [review]
See comments, also you have merge conflicts to deal with

File: src/main/java/org/kpmp/spatialViewerDataset/SpatialViewerDatasetService.java [hunk 4]
```diff
@@ -11,12 +11,15 @@
...
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.io.IOException;
 
+
 @Service
 public class SpatialViewerDatasetService  {
-
+    private static int EQUALITY_OPERATOR = 0;
```
Comment (zwright) [inline src/main/java/org/kpmp/spatialViewerDataset/SpatialViewerDatasetService.java:22]: This is meant to be a constant? You need the "final" keyword too.

**Review by zwright — APPROVED** [review]

**Pull request merged**